src/lem/store.rs

```diff
@@ -161,89 +153,186 @@ impl<F: LurkField> Store<F> {
         self.expect_f(self.hash8zeros_idx)
     }
 
+    /// Converts array of pointers of size `P` to array of raw pointers of size `N` such that `P = N * 2`.
+    /// Since the `generic_const_exprs` feature is still unstable, we cannot substitute `N * 2`
+    /// for generic const `P` and remove it completely, so we must keep it and do a dynamic assertion
+    /// that it equals `N * 2`. This is not very ergonomic though, since we must add turbofishes
+    /// like `::<6, 3>` instead of the simpler `::<3>`. Could we maybe create a macro for these functions?
+    #[inline]
+    pub fn ptrs_to_raw_ptrs<const N: usize, const P: usize>(&self, ptrs: &[Ptr; P]) -> [RawPtr; N] {
+        assert_eq!(P * 2, N);
+        let mut raw_ptrs = [self.raw_zero(); N];
+        for i in 0..P {
+            raw_ptrs[2 * i] = self.tag(*ptrs[i].tag());
+            raw_ptrs[2 * i + 1] = *ptrs[i].raw();
+        }
+        raw_ptrs
+    }
+
+    /// Tries to convert array of raw pointers of size `N` to array of pointers of size `P = N * 2`.
+    /// It might fail since not all raw pointers represent valid tags.
+    #[inline]
+    pub fn raw_ptrs_to_ptrs<const N: usize, const P: usize>(
+        &self,
+        raw_ptrs: &[RawPtr; N],
+    ) -> Option<[Ptr; P]> {
+        assert_eq!(P * 2, N);
```

Comment:
Yeah, `generic_const_exprs` is sorely missing here. 

Thankfully, you're using this for *array operations*, and we already have generic-array in our dependencies. We're therefore in the special case where we can use this obscure but beautiful crate, which I believe is the only way to use const generic array operations with a path to const generic migration when the Rust compiler supports it:
https://docs.rs/hybrid-array/latest/hybrid_array/
https://github.com/RustCrypto/utils/tree/master/hybrid-array